### DISCUSSION TITLE
MapInputName not found in file

### DISCUSSION THREAD

**Discussion by ramtinazimi** [spatie/laravel-data#218] [Q&A] [answered]
New to Laravel and also new to Laravel-Data; the library is awesome!

When I try to use the MapInputName I receive the error:
**MapInputName not found in file.**

This is what my code looks like:

`<?php
namespace App\DTO;

use Spatie\LaravelData\Data;
  
#[MapInputName(SnakeCaseMapper::class)]
class MemberDTO extends Data
{
    public function __construct(

        public string $lastName,
        
        public string $firstName,

       
    ) {
    }
}`

It works if I leave out the MapInputName. Any ideas?

(Sorry for the bad code formating; for some reason I cannot get it work for the whole code be in the code block.)

![image](https://user-images.githubusercontent.com/2991808/191896065-39664caf-e531-4a34-960e-db95e6dcd119.png)

**Comment by rubenvanassche** [comment, marked as answer]
You should import the class: `use Spatie\LaravelData\Attributes\MapInputName;`

**Reply by rognales** [reply]
Hi. I already import ```use Spatie\LaravelData\Attributes\MapOutputName;``` but still getting the error

```console
Attribute class "Spatie\LaravelData\Attributes\MapOutputName" not found
```

Here's my full class (as per docs)

```php
use Spatie\LaravelData\Attributes\MapOutputName;

class ContractData extends Data
{
    public function __construct(
        public string $name,
        #[MapOutputName('record_company')]
        public string $recordCompany,
    ) {
    }
}
```
